### PR TITLE
Mpfs ddr fixes2

### PR DESCRIPTION
**Pull request by jlaitine**
## Summary

A bunch of somewhat tested corrections for the current ddr training code

### CODE REVIEW

File: arch/risc-v/src/mpfs/mpfs_ddr.c [hunk 1]
```diff
@@ -3770,48 +3796,16 @@ static int mpfs_training_full_mtc_test(void)
...
-                         MTC_ADD_RANDOM);
-  error |= mpfs_mtc_test(mask, 0x00, ONE_MB_MTC, MTC_PSEUDO_RANDOM_16BIT,
-                         MTC_ADD_RANDOM);
-  error |= mpfs_mtc_test(mask, 0x00, ONE_MB_MTC, MTC_PSEUDO_RANDOM_8BIT,
-                         MTC_ADD_RANDOM);
+  if (error == 0)
+    {
+      /* Test random additions */
+
+      error |= mpfs_mtc_test_all(mask, 0x00, ONE_MB_MTC, MTC_ADD_RANDOM);
```
Comment (jlaitine) [inline arch/risc-v/src/mpfs/mpfs_ddr.c:3807]: I |= came as as copy-paste, it is cleaner to just have = , I guess

File: arch/risc-v/src/mpfs/mpfs_ddr.c [hunk 2]
```diff
@@ -1898,6 +1898,47 @@ static uint8_t mpfs_mtc_test(uint8_t mask, uint64_t start_address,
   return getreg32(MPFS_DDR_CSR_APB_MT_ERROR_STS) & 0x01;
 }
```
Comment (eenurkka) [inline arch/risc-v/src/mpfs/mpfs_ddr.c:1913]: Missing the header:
/****************************************************************************
 * Name: mpfs_mtc_test_all
 *
 * Description:

Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_ddr.c:1913]: Right, thanks!

**Review by eenurkka — APPROVED** [review]
Looks good

**Pull request merged**